frontend/components/YourWalletPage/WithdrawFunds/WithdrawFundsModal.tsx

```diff
@@ -0,0 +1,186 @@
+import { Button, Flex, Input, Modal, Typography } from 'antd';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { AgentProfile } from '@/components/AgentProfile';
+import { AgentType } from '@/enums/Agent';
+import { useServices } from '@/hooks/useServices';
+
+import { CustomAlert } from '../../Alert';
+import { ShowBalances } from './ShowBalances';
+
+const { Text } = Typography;
+
+const partOfFundsMayBeLockedMessage =
+  "Your agent could have funds in external smart contracts. Withdrawing Pearl funds now means you won't be able to access those agent funds until a Pearl refund and agent rerun. Withdraw funds from your agent's profile first to ensure access.";
+
+const afterWithdrawing =
+  'Your agent will not be able to run again until it is refunded.';
+
+const WithdrawModalSteps = {
+  FUNDS_MAY_BE_LOCKED: 'FUNDS_MAY_BE_LOCKED',
+  SHOW_BALANCES: 'SHOW_BALANCES',
+  WITHDRAW_FUNDS: 'WITHDRAW_FUNDS',
+};
+
+const ToProceedMessage = () => {
+  const { selectedAgentType } = useServices();
+
+  const withdrawMessage = useMemo(() => {
+    switch (selectedAgentType) {
+      case AgentType.PredictTrader:
+        return `This will withdraw all OLAS and XDAI from your account. ${afterWithdrawing}`;
+      case AgentType.AgentsFun:
+        return `This will withdraw all OLAS and ETH from your account. ${afterWithdrawing}`;
+      case AgentType.Modius:
+      case AgentType.Optimus:
+        return `This will withdraw all OLAS, ETH and USDC from your account. ${afterWithdrawing}`;
+      case AgentType.AgentsFunCelo:
+        return `This will withdraw all OLAS and CELO from your account. ${afterWithdrawing}`;
+      default:
+        return `This will withdraw all funds from your account. ${afterWithdrawing}`;
+    }
+  }, [selectedAgentType]);
```

Comment:
could be just a config of tokens for each agent, and then one message where we could do something like
```
const tokens = [...tokensToWithdraw[selectedAgentType]];
if (!tokens) return `This will withdraw all funds from your account. ${afterWithdrawing}`

const lastToken = tokens.pop();
tokensString `${tokens.join(', ')} and ${last}`;

return `This will withdraw all ${tokensString} from your account. ${afterWithdrawing}`

  ```